src/net/parsers/texttracks/webvtt.js

```diff
@@ -0,0 +1,311 @@
+/** 
+* Parse WebVTT from text. Returns an array with:
+* - start : start of current cue, in seconds
+* - end : end of current cue, in seconds
+* - content : HTML formatted cue.
+*
+* Global style is parsed and applied to div element. 
+* Specific style is parsed and applied to class element. 
+*
+* @param {string} text
+* @return {Array<Object>}
+*/
+export default function parseWebVTT(text) {
+  const newLineChar = /\n|\s{2,}/g;
+  const linified = text.split(newLineChar);
+  const cuesArray = [];
+  const styleElements = [];
+
+  if (!linified[0].match(/^WEBVTT.*/)) {
+    console.log("Can't parse WebVTT: Non valid file.");
+    return;
+  }
+
+  let i = 0;
+
+  // Find every style element into WebVTT, and collect style 
+  // associated with class / tags
+  for (i = 1; i<linified.length; i++) {
+    if (!(linified[i].match(/^\r/) || linified[i].length === 0)) {
+      if(findElementType(linified[i]) === "style"){
+        parseStyle(i+1, linified, styleElements);        
+      }
+      while (linified[i] 
+        && !(linified[i].match(/^\r/) || linified[i].length === 0)) {
+        i++;
+      }
+    }
+  }
+
+  // Parse cues, format and apply style.
+  for (i = 1; i < linified.length; i++) {
+    if (!(linified[i].match(/^\r/) || linified[i].length === 0)) {
+      if (findElementType(linified[i]) === "cue") {
+        cuesArray.push(parseCue(i, linified, styleElements));
+      }
+      while (linified[i] 
+        && !(linified[i].match(/^\r/) || linified[i].length === 0)) {
+        i++;
+      }
+    }
+  }
+
+  return cuesArray;
+}
+
+/**
+ * Finds from supposed first line of element 
+ * (WebVTT is line based format) the type of it. It can be:
+ * - note: A comment
+ * - style: A style indication, about all cues, or specific cues.
+ * - cue: The subtitle content.
+ * - region: Informations about wrapping area of subtitle
+ * @param {string} text 
+ */
+function findElementType(text) {
```

Comment:
``findElementType`` seems to be just called with "===", as in "is this line indicating the beginning of a X block?"

It bothers me because we're adding vocabulary here ("note", "style" etc.) for no advantage.

Why not just:
``isStartOfStyleBlock(text)``
``isStartOfCommentBlock(text)``
``isStartOfCueBlock(text)``

isStartOfCueBlock could just do a negation of the isStartOfStyleBlock and isStartOfCommentBlock